test/unit/mriController.test.js

```diff
@@ -0,0 +1,343 @@
+            body: {
+                url: 'abc.com',
+                atlasName: 'MyAtlas',
+                atlasProject: 'Visualisation@',
+                atlasLabelSet: 'SampleLabelSet',
+                token: 'jnqpincpienfcpewnfcpewn123'
+            },
+            value: 0,
+            validationErrors: function() {
+                return;
+            },
+            checkBody: sinon.stub().returns({ notEmpty: reqSpy, isURL: urlSpy })
+        };
+        let resSpy = sinon.spy();
+
+        let res = {
+            status: sinon.stub().returns({ send: sinon.stub().returns({ end: resSpy })}),
+        };
+        await mriController.validatorPost(req, res, () => {});
+        assert.strictEqual(reqSpy.callCount, 1);
+        assert.strictEqual(urlSpy.callCount, 1);
+        assert.strictEqual(resSpy.callCount, 0);
+        sinon.restore();
+    });
+
+    it('download MRI function should download the file when correct input is given', async function() {
+        let req = {
+            db: db,
+            query: {
+                url: 'https://brainbox.pasteur.fr/mri?url=https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz'
+            },
+            dirname: __dirname.split('/test')[0],
+            headers: {},
+            user: {
+                username: ''
+            },
+            session: {
+                returnTo: ''
+            },
+            originalUrl: '',
+            isTokenAuthenticated: true,
+            tokenUsername: '',
+            isAuthenticated: function() {
+                return this.isTokenAuthenticated;
+            },
+            connection: {
+                remoteAddress: 'http://localhost:3000'
+            }
+        };
+        let arr = [];
+        let authenticated = sinon.stub(req, 'isAuthenticated').resolves(true);
+        let res = {
+            render: sinon.spy()
+        };
+        await mriController.mri(req, res);
+        assert.notDeepStrictEqual(authenticated.calledOnce, true);
+        assert.notDeepStrictEqual(res.render.calledOnce, true);
+        sinon.restore();
+    });
+
+    it('reset function() should make the right calls to the DB and work correctly when the URL is correct', async function (){
+        let req = {
+            db: db,
+            query: {
+                url: 'https://brainbox.pasteur.fr/mri?url=https://s3.amazonaws.com/fcp-indi/data/Projects/ABIDE_Initiative/Outputs/freesurfer/5.1/CMU_a_0050642/mri/T1.mgz'
+            }
+        };
+        let resSpy = sinon.spy();
+        let res = {
+            render: sinon.spy(),
+            status: sinon.stub().returns({ send: sinon.stub().returns({ end: resSpy })}),
+        };
+        await mriController.reset(req, res);
+        assert.notDeepStrictEqual(resSpy.calledOnce, true);
+        sinon.restore();
+    });
+
+    it('appMriGet function() should fetch the MRI as directed when the URL is correct', async function() {
```

Comment:
this call may leave a hanging entry in the downloadQueue, which could fail later on. Same problem as in PR #287.